Atomic/rpmwriter.py

```diff
@@ -0,0 +1,369 @@
+import os
+import sys
+import struct
+import subprocess
+import select
+import gzip
+import hashlib
+import tempfile
+import shutil
+
+# More information on the RPM file format can be found here:
+# http://ftp.rpm.org/max-rpm/s1-rpm-file-format-rpm-file-format.html
+# The same documentation was used to initially write this module.
+#
+# Only tags that are actually used in the code were added to the RpmWriter class.
+# The complete list is available here:
+# https://github.com/rpm-software-management/rpm/blob/master/lib/rpmtag.h
+
+class RpmWriter(object):
```

Comment:
I've created a separate repository here: https://github.com/giuseppe/tinyrpmbuild, if someone is really interested in it.

I agree with @cgwalters, this is not a replacement for `rpmbuild`, it is really implementing the minimum necessary to get an installable `.rpm` file.  Not sure how many other users will have the same requirement of building an `.rpm` on the fly without using `rpmbuild`.